applications/HDF5Application/python_scripts/core/file_io.py

```diff
@@ -97,6 +110,18 @@ def Get(self, model_part=None):
             filename = filename.replace('<model_part_name>', model_part.Name)
         if not filename.endswith('.h5'):
             filename += '.h5'
+
+        if (not self.list_of_file_names is None):
+            try:
+                pos = self.list_of_file_names.index("")
+                self.list_of_file_names[pos] = filename
+            except ValueError:
+                first_file_name = self.list_of_file_names.pop(0)
+                if hasattr(model_part, "GetCommunicator"):
+                    if (model_part.GetCommunicator().MyPID() == 0):
+                        kratos_utils.DeleteFileIfExisting(first_file_name)
```

Comment:
```suggestion
                kratos_utils.DeleteFileIfExisting(first_file_name)
```

this works in mpi too

Reply:
done :)